.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
```

Comment:
should we add another hook for python linter? Then you do not need to wait until CI runs through and clippy for rust files maybe. Just a thought

Reply:
flake8 and pylint has similar checks, but with some differences. At the moment we do not do any linter checks with pylint/mypy in CI, but in the long run we better should. In feeder we use pylint/mypy, but only fail on errors. I did by the way try mypy some time ago on kuksa-client but it gave some false positives as it could not really understand the gRPC integration correctly, so there are some tweaks needed if we are to use it.

It is possiblw to run pylint as part of pre-commit, see https://pylint.pycqa.org/en/latest/user_guide/installation/pre-commit-integration.html, but it requires that you do a local install. That is no problem for us working with Python but I am not sure if our rust/c++-guys want to do that, as it might be needed even if you have not changed any Python files.

I suggest we take extending/improving checks as a later step.